php/admin/class-migration.php

```diff
@@ -0,0 +1,207 @@
+						esc_html__( 'The Block Lab team have moved. For future updates and improvements, migrate now to the new home of custom blocks: %1$s.', 'block-lab' ),
+						'<strong>Genesis Custom Blocks</strong>'
+					);
+					?>
+				</p>
+				<p>
+					<?php
+					printf(
+						'<a target="_blank" href="https://getblocklab.com/docs/genesis-custom-blocks">%1$s</a>',
+						esc_html__( 'Learn more', 'block-lab' )
+					);
+					?>
+				</p>
+			</div>
+			<a href="#" id="bl-notice-not-now" class="bl-notice-option">
+				<?php esc_html_e( 'Not Now', 'block-lab' ); ?>
+			</a>
+			<a href="<?php echo esc_url( $migration_url ); ?>" class="bl-notice-option">
+				<?php esc_html_e( 'Migrate', 'block-lab' ); ?>
+			</a>
+		</div>
+		<div id="bl-not-now-notice" class="notice notice-info bl-notice-migration bl-hidden">
+			<div class="bl-migration-copy">
+				<p><?php esc_html_e( "When you're ready, our migration tool is available in the main menu, under Block Lab > Migrate.", 'block-lab' ); ?></p>
+			</div>
+			<a href="#" id="bl-notice-ok" class="bl-notice-option">
+				<?php esc_html_e( 'Okay', 'block-lab' ); ?>
+			</a>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Enqueues the migration notice assets.
+	 */
+	public function enqueue_assets() {
+		if ( ! $this->should_display_migration_notice() ) {
+			return;
+		}
+
+		wp_enqueue_style(
+			self::NOTICE_STYLE_SLUG,
+			$this->plugin->get_url( 'css/admin.migration-notice.css' ),
+			[],
+			$this->plugin->get_version()
+		);
+
+		wp_enqueue_script(
+			self::NOTICE_SCRIPT_SLUG,
+			$this->plugin->get_url( 'js/admin.migration-notice.js' ),
+			[],
+			$this->plugin->get_version(),
+			true
+		);
+	}
+
+	/**
+	 * Gets whether the migration notice should display.
+	 *
+	 * This should display on Block Lab > Content Blocks,
+	 * /wp-admin/plugins.php, the Dashboard, and Block Lab > Settings.
+	 *
+	 * @return bool Whether the migration notice should display.
+	 */
+	public function should_display_migration_notice() {
+		if ( ! current_user_can( self::NOTICE_CAPABILITY ) ) {
+			return false;
+		}
+
+		// If the user has dismissed the notice, it shouldn't appear again.
+		if ( self::NOTICE_DISMISSED_META_VALUE === get_user_meta( get_current_user_id(), self::NOTICE_USER_META_KEY, true ) ) {
+			return false;
+		}
+
+		$screen = get_current_screen();
+		return (
+			( isset( $screen->base, $screen->post_type ) && 'edit' === $screen->base && 'block_lab' === $screen->post_type )
+			||
+			( isset( $screen->base ) && in_array( $screen->base, [ 'plugins', 'dashboard', 'block_lab_page_block-lab-settings' ], true ) )
+		);
+	}
+
+	/**
+	 * Handles an AJAX request to not display the notice.
+	 *
+	 * This stores in the user meta the fact that the notice was dismissed,
+	 * so it's not displayed again.
+	 */
+	public function ajax_handler_migration_notice() {
+		check_ajax_referer( self::NOTICE_NONCE_ACTION, self::NOTICE_NONCE_NAME );
+
+		if ( ! current_user_can( self::NOTICE_CAPABILITY ) ) {
+			wp_die( -1 );
```

Comment:
Why use `wp_die()` over [`wp_send_json_error()`](https://developer.wordpress.org/reference/functions/wp_send_json_error/)?

Reply:
Good question, I'm fixing it 😄

Reply:
Fixed in https://github.com/getblocklab/block-lab/pull/635/commits/d2ce757313994d0d487d55cbc9c17afe8d9f4e95